### PR TITLE
MSP improvements

### PR DESCRIPTION
**Pull request by digitalentity**
Minor refactoring of `MSP_SET_4WAY_IF` handling. 
Allow `MSP_DATAFLASH_READ` to specify desired read (payload) size.

New `MSP_DATAFLASH_READ` request format:
```
uint32_t - read address
uint16_t - read size (optional)
```

Response payload format didn't change.

### CODE REVIEW

**Comment by digitalentity** [comment]
MSPv2 part tested. Testing of BLHeli passthrough is needed

File: src/main/fc/fc_core.c [hunk 1]
```diff
@@ -269,11 +269,6 @@ static void updateArmingStatus(void)
 
         warningLedUpdate();
     }
-
-    debug[0] = (armingFlags >> 0) & 0x00FF;
-    debug[1] = (armingFlags >> 8) & 0x00FF;
-    debug[2] = (armingFlags >> 16) & 0x00FF;
-    debug[3] = (armingFlags >> 24) & 0x00FF;
 }
```
Comment (fiam) [inline src/main/fc/fc_core.c:272]: This seems unrelated to the changes mentioned in the PR. Was it accidental?

File: src/main/msp/msp_serial.h [hunk 2]
```diff
@@ -83,11 +83,11 @@ typedef struct mspPort_s {
     struct serialPort_s *port; // null when port unused.
     mspState_e c_state;
     uint8_t inBuf[MSP_PORT_INBUF_SIZE];
-    uint16_t cmdMSP;
-    uint8_t cmdFlags;
+    int offset;
+    int dataSize;
     mspVersion_e mspVersion;
```
Comment (fiam) [inline src/main/msp/msp_serial.h]: Why not fixed width types instead of int for `offset` and `dataSize`? From the code, it seems that they both would always fit in an uint16_t.

File: src/main/fc/fc_core.c (hunk 1 above)
Comment (digitalentity) [reply src/main/fc/fc_core.c:272]: This is accidental, but should be merged anyway.

File: src/main/msp/msp_serial.h (hunk 2 above)
Comment (digitalentity) [reply src/main/msp/msp_serial.h]: Using a generic type will allow compiler to optimize better. Using a fixed data type might force compiler to enforce that type by using narrow instructions.

Comment (fiam) [reply src/main/msp/msp_serial.h]: If you want the compiler to optimize for speed there, I'd use `uint_fast16_t`. I guess we're unlikely to start supporting platforms where int is 16 bits so it should be fine either way, but I'd make them at least unsigned for readability since they are both only assigned to unsigned values.

Comment (digitalentity) [reply src/main/msp/msp_serial.h]: Good point

**Comment by digitalentity** [comment]
BLHeli passthrough tested
